demo/api/src/products/entities/product-category.entity.ts

```diff
@@ -23,6 +23,10 @@ export class ProductCategory extends BaseEntity<ProductCategory, "id"> {
     @Field()
     slug: string;
 
+    @Property({ columnType: "integer" })
+    @Field(() => Int)
+    position: number;
```

Comment:
This should have `@Min(1)` to ensure positive numbers only and make it directly visible that position starts with 1

Reply:
this field is probably excluded from create and update mutation so I don't know if this will have any impact. But at least it's a hint how this field is used.

Reply:
Validators should also be applied to inputs. I think this was added to the API generator. @nsams?

Reply:
yes, this was recently added

Reply:
I meant the position field should not be part of create-input (fixed set to last or first position) nor update-input (because this would require the same updatePosition code like in the updatePosition-mutation)

should I prefer defining an extra DTO to replace the `if (position < 1) throw CometValidationException` with a class-validator check?

Reply:
I think you don't understand what we are talking about :)

This PR https://github.com/vivid-planet/comet/pull/1828 adds a fature that copies validators from entity into input dtos.

you either didn't run the generator after adding the validator, or it doesn't work correctly in your case